docs/testing.md

```diff
@@ -247,51 +247,86 @@ Jest and React Testing Library are frequently used together for **Unit Testing**
 
 ### Quickstart
 
-You can use `create-next-app` with the [with-jest example](https://github.com/vercel/next.js/tree/canary/examples/with-jest) to quickly get started with Jest and React Testing Library:
+You can use `create-next-app` with the [with-jest](https://github.com/vercel/next.js/tree/canary/examples/with-jest) example to quickly get started with Jest and React Testing Library:
 
 ```bash
 npx create-next-app@latest --example with-jest with-jest-app
 ```
 
-### Manual setup
+### Setting up Jest (with SWC)
+
+Since the release of (Next.js 12)[https://nextjs.org/blog/next-12], Next.js now has built-in configuration for Jest. This includes TypeScript support, auto mocking of stylesheets and images, and more.
 
-To manually set up Jest and React Testing Library, install `jest` , `@testing-library/react`, `@testing-library/jest-dom` as well as some supporting packages:
+To set up Jest, install `jest` , `@testing-library/react`, `@testing-library/jest-dom` and `react-test-renderer`.
 
 ```bash
-npm install --save-dev jest babel-jest @testing-library/react @testing-library/jest-dom identity-obj-proxy react-test-renderer
+npm install --save-dev jest @testing-library/react @testing-library/jest-dom react-test-renderer
 ```
 
-**Configuring Jest**
-
-Create a `jest.config.js` file in your project's root directory and add the following configuration options:
+Create a `jest.config.js` file in your project's root directory and add the following:
 
 ```jsx
 // jest.config.js
+const nextJest = require('next/jest')
+
+const createJestConfig = nextJest({
+  // Provide the path to your Next.js app to load next.config.js and .env files in your test environment
+  dir: './',
+})
+
+// Add any custom config to be passed to Jest
+const customJestConfig = {
+  setupFilesAfterEnv: ['<rootDir>/jest.setup.js'],
+}
+
+// createJestConfig is exported this way to ensure that next/jest can load the Next.js config which is async
+module.exports = createJestConfig(customJestConfig)
+```
+
+Under the hood, `next/jest` is automatically configuring Jest for you, including:
+
+- Auto mocking stylesheets (.css, .module.css, and their scss variants) and image imports
+- Setting up `transform` using SWC
+- Loading .env (and all variants) into process.env
+- Ignoring node_modules from test resolving and transforms
+- Ignoring .next from test resolving
+- Loading next.config.js for flags that enable experimental SWC transforms
+
+### Setting up Jest (with Babel)
 
+If you opt-out of SWC and use Babel instead. In addition to packages above, you will need to install the `babel-jest` and `identity-obj-proxy` packages, and manually configure Jest.
```

Comment:
Recommend using "opt out" instead of "opt-out" based on the context. Also, "If you opt-out of SWC and use Babel instead." is an incomplete sentence – did you mean to add something?

Reply:
Hm, yes, I think you're right. That doesn't read smoothly, does it?

Reply:
Maybe, "If you opt out of SWC, use Babel instead" would work better? I don't think that "and" is supposed to be there :)

Reply:
How about? 

> If you opt out of SWC, you will need to manually configure Jest and install `babel-jest` and `identity-obj-proxy` in addition to the packages above.

Reply:
Perfect!

Reply:
ty!